jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Utf8Appendable.java

```diff
@@ -327,40 +331,159 @@ public boolean isUtf8SequenceComplete()
     }
 
     /**
-     * Not a valid sequences of UTF-8 bytes
+     * Finish the buffer state.
+     * <p>
+     *     This will address any incomplete utf-8 sequences in the buffer, and set Utf8Appendable in a state
+     *     where it will be ready to accept new utf-8 sequences.
+     * </p>
      */
-    public static class NotUtf8Exception extends IllegalArgumentException
+    public void finish()
```

Comment:
this can probably be private now (I like the name better)

Reply:
I gave this a whirl.

The websocket ValidationExtension still needs this. (to avoid a String copy).
Once we have a solution for that, then we can go `protected` for this field, as the Utf8StringBuilder and Utf8StringBuffer still use it.